### PR TITLE
feat(datasets): JSON endpoint to get dataset versions

### LINKED ISSUE #3215: [datasets] GET /v1/datasets/:dataset_id/versions/

### PR DESCRIPTION
**Pull request by RogerHYang**
resolves #3215

### CODE REVIEW

File: src/phoenix/server/api/routers/v1/datasets.py [hunk 1]
```diff
@@ -213,6 +213,118 @@ async def get_dataset_by_id(request: Request) -> Response:
...
+                status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+    else:
+        return Response(
+            content="Missing Dataset ID",
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+        )
+    try:
+        limit = int(request.query_params.get("limit", 10))
+        assert limit > 0
```
Comment (axiomofjoy) [inline src/phoenix/server/api/routers/v1/datasets.py:285]: 422 might be appropriate here.

File: tests/server/api/routers/v1/test_datasets.py [hunk 2]
```diff
@@ -188,6 +188,66 @@ async def test_list_datasets_with_cursor(
     assert all(item in second_page_datasets[0].items() for item in fixture_values.items())
 
 
+async def test_get_dataset_versions(test_client, dataset_with_revisions):
+    dataset_global_id = GlobalID("Dataset", str(2))
+    response = await test_client.get(f"/v1/datasets/{dataset_global_id}/versions?limit=2")
+    assert response.status_code == 200
+    assert response.headers.get("content-type") == "application/json"
+    assert response.json() == {
+        "next_cursor": "RGF0YXNldFZlcnNpb246Nw==",
```
Comment (axiomofjoy) [inline tests/server/api/routers/v1/test_datasets.py:197]: Suggestion: It would help readability in the tests to use `str(GlobalID(...))` instead of the raw cursor value.

File: tests/server/api/routers/v1/test_datasets.py [hunk 3]
```diff
@@ -188,6 +188,66 @@ async def test_list_datasets_with_cursor(
...
+
+async def test_get_dataset_versions_with_cursor(test_client, dataset_with_revisions):
+    dataset_global_id = GlobalID("Dataset", str(2))
+    response = await test_client.get(
+        f"/v1/datasets/{dataset_global_id}/versions?limit=2&cursor=RGF0YXNldFZlcnNpb246Nw=="
+    )
+    assert response.status_code == 200
+    assert response.headers.get("content-type") == "application/json"
+    assert response.json() == {
+        "next_cursor": "RGF0YXNldFZlcnNpb246NQ==",
```
Comment (axiomofjoy) [inline tests/server/api/routers/v1/test_datasets.py:223]: Suggestion: Might be worth having one test that queries right up until the end so `next_cursor` is `None`.

File: tests/server/api/routers/v1/test_datasets.py [hunk 4]
```diff
@@ -188,6 +188,66 @@ async def test_list_datasets_with_cursor(
...
+        ],
+    }
+
+
+async def test_get_dataset_versions_with_invalid_cursor(test_client, dataset_with_revisions):
+    dataset_global_id = GlobalID("Dataset", str(2))
+    response = await test_client.get(
+        f"/v1/datasets/{dataset_global_id}/versions?limit=1&cursor=RGF0YXNldFZlcnNpb246MQ=="
+    )
+    assert response.status_code == 200
```
Comment (axiomofjoy) [inline tests/server/api/routers/v1/test_datasets.py:246]: Wouldn't we want to return a 4xx status code in this case?

**Review by axiomofjoy — APPROVED** [review]

File: tests/server/api/routers/v1/test_datasets.py (hunk 4 above)
Comment (RogerHYang) [reply tests/server/api/routers/v1/test_datasets.py:246]: It's actually a version from a different dataset, so it's not 422-invalid, but it's an edge case, so i didn't add the extra query for it.

File: src/phoenix/server/api/routers/v1/datasets.py (hunk 1 above)
Comment (RogerHYang) [reply src/phoenix/server/api/routers/v1/datasets.py:285]: That's on line 289
